onetl/hwm/auto_hwm.py

```diff
@@ -11,11 +11,39 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
+from __future__ import annotations
 
+from typing import Any
 
-from etl_entities.hwm import ColumnHWM
+from etl_entities.hwm import HWM
+from pydantic import root_validator
 from typing_extensions import Literal
 
 
-class AutoDetectHWM(ColumnHWM):
+class AutoDetectHWM(HWM):
     value: Literal[None] = None
+
+    @root_validator(pre=True)
+    def handle_aliases(cls, values):
+        # this validator is hack for accommodating multiple aliases for a single field in pydantic v1.
+
+        # 'column' is an alias used specifically for instances of the ColumnHWM class.
+        if "source" in values and "entity" not in values:
+            values["entity"] = values.pop("source")
+
+        # 'topic' is an alias used for instances of the KeyValueHWM class.
+        elif "topic" in values and "entity" not in values:
+            values["entity"] = values.pop("topic")
+
+        return values
+
+    def update(self: AutoDetectHWM, value: Any) -> AutoDetectHWM:
+        """Update current HWM value with some implementation-specific logic, and return HWM"""
+        return self
```

Comment:
```suggestion
        raise NotImplementedError()
```
На случай, если мы где-то пропустим, и AutoDetectHWM попадет внутрь какой-нибудь стратегии